src/libs/elektra/hooks.c

```diff
@@ -0,0 +1,95 @@
+/**
+ * @file
+ *
+ * @brief
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ */
+
+#include <kdbinternal.h>
+
+void freeHooks (KDB * kdb, Key * errorKey)
+{
+	if (kdb->hooks == NULL)
+	{
+		return;
+	}
+
+	if (kdb->hooks->gopts != NULL)
+	{
+		elektraPluginClose (kdb->hooks->gopts->plugin, errorKey);
+		free (kdb->hooks->gopts);
+		kdb->hooks->gopts = NULL;
+	}
+
+	free (kdb->hooks);
+	kdb->hooks = NULL;
+}
+
+static struct _HookPluginGopts * initHooksGopts (Plugin * plugin)
+{
+	if (!plugin)
+	{
+		return NULL;
+	}
+
+	struct _HookPluginGopts * hook = elektraCalloc (sizeof (struct _HookPluginGopts));
+	hook->plugin = plugin;
+
+	hook->kdbHookGoptsGet = (kdbHookGoptsGetPtr) elektraPluginGetFunction (plugin, "hooks/gopts/get");
```

Comment:
You should also check that this function exists (`!= NULL`) and if not produce an error/warning.